### PR TITLE
[Test] Integration tests

### LINKED ISSUE #152: [Test] Integration Tests for lib/analysis/author/
**Description**
- [x] community.py
- [ ] ranking.py
- [x] edge_list.py
- [x] time_statistics.py
- [x] wh_table.py
- [x] curve_fitting.py

In graph/
- [x] bipartite.py
- [x] generate.py
- [x] interaction.py

### LINKED ISSUE #155: [Test] Integration Tests for lib/input/
**Description**
- [x] check_headers.py
- [x] data_cleanup.py
- [ ] ps_table.py

In mbox/
- [ ] keyword_clustering.py
- [ ] keyword_digest.py
- [ ] mbox_hdr.py

In imap/
- [ ] connection.py
- [ ] header.py

### PR DESCRIPTION
**Pull request by hrishikesh-dahiya**
## Description
 - Integrations tests for `generate.py`, `keyword_clustering.py`, `data_cleanup.py`
 - Added params to generate.py and add the documentation as well
 - Fixes #155  and #152

## Motivation
 - To increase the code coverage
 
 ## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checks
 - [x] Squash your commits down to one, or a few, discreet changesets before submitting a pull request
 - [x] Read the [contributing guide](https://github.com/DeveloperCAP/MLCAT/blob/development/.github/CONTRIBUTING.md)
 - [x] PR should be motivated, i.e. what does it fix, why, and if relevant how
 - [ ] If possible, include an example in the description, that could help all readers (including project members) to get a better picture of the change
 - [x] Avoid other runtime dependencies unless absolutely necessary 
 - [x] Rebase your commit history so that each commit is meaningful and states the intention behind the change 
 - [x] Ensure that your pull request follows the [style guidelines](https://github.com/DeveloperCAP/MLCAT/wiki/Coding-Guidelines)
 - [x] At least one commit should mention `Fixes #152 and #155` _if relevant_

## Notify reviewers
@prasadtalasila

### CODE REVIEW

**Comment by hrishikesh-dahiya** [comment]
@prasadtalasila any solution to codeclimate [issue](https://codeclimate.com/github/DeveloperCAP/MLCAT/pull/158) for now? Because we need all 5 param for testing it. I think refactoring work can be done later.

File: lib/analysis/author/graph/generate.py [hunk 1]
```diff
@@ -70,8 +74,8 @@ def author_interaction():
 					json_data[json_obj['Message-ID']] = json_obj
 		print("JSON data loaded.")
 	else:
-		lone_author_threads = get_lone_author_threads(False)
-		with open('clean_data.json', 'r') as json_file:
+		lone_author_threads = get_lone_author_threads(nodelist_filename=graph_nodes, edgelist_filename=graph_edges)
```
Comment (prasadtalasila) [inline lib/analysis/author/graph/generate.py:76]: please use positional arguments here.

File: test/integration_test/data/headers_cl.json [hunk 2]
```diff
@@ -0,0 +1,279 @@
+{
```
Comment (prasadtalasila) [inline test/integration_test/data/headers_cl.json]: short test data files are preferable to long ones. If you can reduce the number of entries in this file and still have the test coverage, that would be a good thing.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]

File: lib/analysis/author/graph/generate.py (hunk 1 above)
Comment (hrishikesh-dahiya) [reply lib/analysis/author/graph/generate.py:76]: `get_lone_author_threads()` is a function of read.py, which was to be refactored afterward.

**Comment by hrishikesh-dahiya** [comment]
Data input size reduced for `test_data_cleanup.py`.
nx.is_isomorphic(G1, G2) used in `test_generate.py` for asserting two graphs.
nx.read_pajek(filename) used to read a .net file to a graph.
Previous tests' output files added for reference.
Travis CI Build Passing, same codeclimate issue(to be refactored later).

File: lib/analysis/author/graph/generate.py [hunk 3]
```diff
@@ -35,16 +35,20 @@ def write_to_pajek(author_graph, filename="author_graph.net"):
...
+	:param clean_data: Path to clean_data.json file
+	:param graph_nodes: Path to graph_nodes.csv file
+	:param graph_edges: Path to graph_edges.csv file
+	:param pajek_file: Path to author_graph.net(pajek file) which is used by write_to_pajek()
+	:param ignore_lat: True
 	"""
 	# Time limit can be specified here in the form of a timestamp in one of the identifiable formats and all messages
 	# that have arrived after this timestamp will be ignored.
 	time_limit = None
 	# If true, then messages that belong to threads that have only a single author are ignored.
```
Comment (prasadtalasila) [inline lib/analysis/author/graph/generate.py]: please move this description to sphinx docs of `ignore_lat` variable.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
@hrishikesh-dahiya one minor change in the sphinx docs

**Review by prasadtalasila — APPROVED** [review]

**Pull request merged**